client/src/setup/setupCommands.ts

```diff
@@ -0,0 +1,31 @@
+import * as vscode from "vscode";
+import { ExtensionState } from "../types";
+
+import CompileCommand from "../commands/CompileCommand";
+import FlattenCurrentFileCommand from "../commands/FlattenCurrentFileCommand";
+import CleanCommand from "../commands/CleanCommand";
+
+const commandClasses = [
+  CompileCommand,
+  FlattenCurrentFileCommand,
+  CleanCommand,
+];
+
+export function setupCommands(state: ExtensionState) {
+  for (const commandClass of commandClasses) {
+    const command = new commandClass(getOutputChannel(state));
+    const disposable = vscode.commands.registerCommand(
+      `hardhat.solidity.${command.name()}`,
+      () => command.execute()
+    );
+    state.context.subscriptions.push(disposable);
+  }
+}
+
+const getOutputChannel = (state: ExtensionState) => {
+  if (!state.commandsOutputChannel) {
+    state.commandsOutputChannel =
+      vscode.window.createOutputChannel("Hardhat Commands");
```

Comment:
I don't think so, but I thought it would be better to have it separated from extension general logging. Should I use the 'Hardhat' channel for commands output?

Reply:
I am not sure. I can see that the output is pretty different from logging, I am just unsure whether the surprise in finding it not in the hardhat channel but a different channel out weights it.
Did I understand correctly, if there is an error in the command, the output channel is shown?

Reply:
Yes, so when you run let's say the Compile command, a progress bar is shown, and hidden once it finishes. If there are no errors, then the CLI output is not shown (I think it's not important). But if it fails, then the CLI output is shown. What do you think?

Reply:
Alright, lets keep the "Hardhat Commands" channel. If the only time you are really going to encounter it is when we show it to you, then the _oddness_ of a different channel is not that important.